canlink/README.md

```diff
@@ -1,78 +1,98 @@
+
+The bus manager is designed to be the __single point of contact__ to a can interface. All interaction to a bus should be done through the bus manager.
 
-The CAN Tracer captures all traffic on a CAN bus 
-during a test and dumps it into a file.
 
 ### Usage
 
-1) Create an instance of Tracer with the _NewTracer()_ function. 
-A can interface, directory and logger must be provided as arguments. 
-Functional options are available of type _TracerOption_ if required. 
+1) Create a Bus Manager using `NewBusManager()` function.
+A logger, and pointer to a socketcan connection are passed as arguments.
+
+    ```go
+    func main() {
+        ctx := context.Background()
+
+        loggerConfig := zap.NewDevelopmentConfig()
+        logger, err := loggerConfig.Build()
+
+        // Create a network connection for vcan0
+        conn, err := socketcan.DialContext(context.Background(), "can", "vcan0")
+        if err != nil {
+            return
+        }
+
+        manager := canlink.NewBusManager(logger, &conn)
+    }
+    ```
+
+2) Create an instance of Tracer with the `NewTracer()` function. 
+A can interface, logger, and converter must be provided as arguments. 
+A timeout and a file name are optional parameters. If no file name is provided, a name will be generated using the current time and date. The tracer will timeout after 30 minutes by default. The trace file will be created in the same directory.
+Functional options are available of type `TracerOption` if required. 
 
     ```go
     func main() {
         tracer := canlink.NewTracer(
-            "can0",
-            "/opt/traces",
+            "vcan0",
             logger,
-            canlink.WithBusName("PT"),
-            canlink.WithTimeout(5*time.Minute))
+            canlink.Text{}
+            canlink.WithTimeout(1*time.Second)
+            canlink.WithFileName("trace_sample")
+	    )
     }
     ```
-2) Open the Tracer using _Open()_
+3) Register the `Tracer` instance as a handler for the bus manager by calling `Register`, passing in the `Tracer`. Once started, the manager will send frames from the CAN bus through the broadcast channel to every registered handler.
 
     ```go
     func main() {
-        err = tracer.Open(ctx)
+        broadcast := manager.Register(tracer)
+        go tracer.Handle(broadcast)
+	    defer tracer.Close()
     }
     ```
-3) To start the tracer and start recording traffic, call _StartTrace()_
+4) Then call `tracer.Handle` on a separate go routine with the broadcast channel as an argument. Once `tracer.Handle` is called any frames transmitted on the broadcast channel will be traced. 
```

Comment:
It there ever a case where you would register a handler but not want the separate go routine for Handle? I think they always occur together, so can this step automatically happen in a previous step?